modules/perception/camera/lib/motion/plane_motion.cc

```diff
@@ -56,13 +56,10 @@ void PlaneMotion::generate_motion_matrix(base::VehicleStatus *vehicledata) {
         sqrt(vehicledata->velocity_x * vehicledata->velocity_x +
              vehicledata->velocity_y * vehicledata->velocity_y));
     float displacement = time_d * velocity;
-    trans(0) = static_cast<float>(displacement * cos(theta));
-    trans(1) = static_cast<float>(displacement * sin(theta));
-    // trans(0) = time_d * vehicledata->velocity_x;
-    // trans(1) = time_d * vehicledata->velocity_y;
+    trans << displacement, 0;
```

Comment:
seems keeping the original code is more generic.

Reply:
Here, it is equivalent.

Reply:
Just want to keep consistent with the code style of `motion_3d`.
Maybe, we could comment it, not delete.
Or, code the `motion_3d` part in a complex way like `motion_2d` of the original. But I have not figured out have to calculate it.
In a word, this is a simple and easy solution I can figure out.

Reply:
The modified code is incorrect.

Reply:
@techoe 
Absolutely, you are right! It is my fault:)
Please help clarify the doubt. 
The code snip 
```
    trans(0) = static_cast<float>(displacement * cos(theta));
    trans(1) = static_cast<float>(displacement * sin(theta));
```
and 
```
    // trans(0) = time_d * vehicledata->velocity_x;
    // trans(1) = time_d * vehicledata->velocity_y;
```

is equivalent, right?

If so, we code use the latter one to format the motion 3D, which should be more accurate than the original one
```
    trans(0) =
        static_cast<float>(sqrt(displacement * displacement /
                                (tan(yaw_delta) * tan(yaw_delta) +
                                 tan(pitch_delta) * tan(pitch_delta) + 1)));
    trans(1) = static_cast<float>(tan(yaw_delta) * trans(0));
    trans(2) = static_cast<float>(tan(pitch_delta) * trans(0));
```